packages/embedder/indexing/retrospectiveDiscussionTopic.ts

```diff
@@ -0,0 +1,267 @@
+//  - <PARTICIPANT|ANONYMOUS> wrote, "<RESPONSE>"
+//
+// </FOR EACH PROMPT>
+//
+// <IF DISCUSSION>
+// A discussion was held. <DISCUSSION SUMMARY>
+// </END IF DISCUSSION>
+
+const IGNORE_COMMENT_USER_IDS = ['parabolAIUser']
+
+const pg = getKysely()
+
+export async function refreshEmbeddingsMeta(dataLoader: DataLoaderWorker) {
+  const r = await getRethink()
+  const {createdAt: newestMeetingDate} = (await r
+    .table('NewMeeting')
+    .max({index: 'createdAt'})
+    .run()) as unknown as RethinkSchema['NewMeeting']['type']
+  const {createdAt: oldestMeetingDate} = (await r
+    .table('NewMeeting')
+    .min({index: 'createdAt'})
+    .run()) as unknown as RethinkSchema['NewMeeting']['type']
+
+  const {newestIndexDate} = (await pg
+    .selectFrom('EmbeddingsMetadata')
+    .select(pg.fn.max('refUpdatedAt').as('newestIndexDate'))
+    .executeTakeFirst()) ?? {newestIndexDate: null}
+  const startDateTime = newestIndexDate || oldestMeetingDate
+  console.log(`Index will consider adding items from ${startDateTime} to ${newestMeetingDate}`)
+
+  if (startDateTime.getTime() === newestMeetingDate.getTime()) return
+
+  for (const interval of generateHourIntervals(startDateTime, newestMeetingDate)) {
+    // fetch completed meetings within the time range, ensuring these meetings
+    // contain at least 1 discuss phase
+    const meetings = await r
+      .table('NewMeeting')
+      .between(interval[0], interval[1], {rightBound: 'closed', index: 'createdAt'})
+      .filter((r: RDatum) =>
+        r('meetingType')
+          .eq('retrospective')
+          .and(
+            r('summarySentAt').gt(0),
+            r
+              .hasFields('phases')
+              .and(r('phases').count().gt(0))
+              .and(
+                r('phases')
+                  .filter((phase: RDatum) => phase('phaseType').eq('discuss'))
+                  .filter((phase: RDatum) =>
+                    phase.hasFields('stages').and(phase('stages').count().gt(0))
+                  )
+                  .count()
+                  .gt(0)
+              )
+          )
+      )
+      .run()
+    const embeddingsMetaRows = (
+      await Promise.all(meetings.map((m) => newRetroDiscussionTopicsFromNewMeeting(m, dataLoader)))
+    ).flat()
+    if (!embeddingsMetaRows.length) continue
+    await upsertEmbeddingsMetaRows(embeddingsMetaRows)
+  }
+  return
+}
+
+async function getPreferredNameByUserId(userId: string, dataLoader: DataLoaderWorker) {
+  const user = await dataLoader.get('users').load(userId)
+  return !user ? 'Unknown' : user.preferredName
+}
+
+async function formatThread(
+  dataLoader: DataLoaderWorker,
+  comments: Comment[],
+  parentId: string | null = null,
+  depth = 0
+): Promise<string> {
+  // Filter and sort comments as before
+  const filteredComments = comments
+    .filter((comment) => comment.threadParentId === parentId)
+    .sort((a, b) => a.threadSortOrder - b.threadSortOrder)
```

Comment:
-1 `a.threadSortOrder < b.threadSortOrder ? -1 : 1`. shouldn't be an issue here, but it will better support NaN. back in the day safari would barf if you didn't write it like this